### ISSUE TITLE
Page numbers for PDF export

### ISSUE THREAD

**Issue by fulldecent** [typora/typora-issues#608]
Hello, I love Typora and use it whenever possible. But it is unacceptable for delivering documents to parties outside our organization because they always expect page numbers.

Please add page numbers as a printing option.

---

I considered doing this as custom CSS. However, this is not supported and won't be supported for a while in WebKit http://caniuse.com/#feat=css-paged-media

Therefore I recommend implementing this as a hack in postscript.

If you are using PanDoc, there may be a trick where you can insert latex commands into markdown before exporting to PDF. Some discussion here http://stackoverflow.com/questions/17232677/disable-page-numbering-converting-markdown-to-pdf-with-pandoc

**Comment by lloyd094** [comment]
I too would like to have this feature, as using Typora allows for much nicer looking documents!

**Comment by abnerlee** [comment]
may relates to https://github.com/electron/electron/issues/7849

**Comment by jolenzy** [comment]
Yes, having this feature would be really great indeed. Please make it so :)

**Comment by zobbo** [comment]
Another bump for me. 

Being able to customise headers and footers would be awesome but I could go a long way with just being able to add page numbers.

I absolutely adore Typora by the way - great work and thank you.

**Comment by abnerlee** [comment]
relates https://github.com/typora/typora-issues/issues/256

**Comment by mandykaandorp** [comment]
Hi, 

I really like Typora and created a nice CSS for formatting, but after exporting my Markdown to PDF I am missing pagenumbers. I have not found a solution in Typora so I created a little script to add pagenumbers to the PDF file. If you want to use it: https://github.com/mandykaandorp/pdffooter (downside: it breaks links in the PDF).

I would really like it if Typora would support pagenumbers (preferably also in the table of contents). 
Regards.

**Comment by cupnoodle** [comment]
My Apologies if this sounds off topic, feel free to remove this comment if it is.

Hi guys, I wrote multiple ebooks using Typora and loved its export to PDF function, I have also faced the same problem of not having page numbers option.

If you are using macOS, I have made a macOS app for adding page numbers on the exported PDF files.
You can download it here [https://pdfpagenumber.com/](https://pdfpagenumber.com/) (Use the discount code 'typorarocks' to get it for free).

Hope this helps!

**Comment by adamgranthendry** [comment]
I would like to have this feature as well

**Comment by Sieboldianus** [comment]
Why was this closed? As far as I can see, there is still no way to include page numbers in the PDF export through typora in 2025?

**Comment by zobbo** [comment]
@Sieboldianus Aye, I stopped using Typora for this reason. Seems a small thing but ended up being a deal breaker for me.

**Comment by drizmiz** [comment]
> Why was this closed? As far as I can see, there is still no way to include page numbers in the PDF export through typora in 2025?

I find that there are at least two ways. 1. https://support.typora.io/Export/#header--footer 2. Use YAML
```{yaml}
---
footer: Page ${pageNo} from ${totalPages}
---
```
Ref: #118
